Include/object.h

```diff
@@ -140,8 +140,12 @@ static inline int _Py_IS_TYPE(const PyObject *ob, const PyTypeObject *type) {
 }
 #define Py_IS_TYPE(ob, type) _Py_IS_TYPE(_PyObject_CAST_CONST(ob), type)
 
-
 static inline void _Py_SET_REFCNT(PyObject *ob, Py_ssize_t refcnt) {
+    // We don't have clean access to pycore_object.h at this point.
+    PyAPI_FUNC(int) _PyObject_IsImmortal(PyObject *);
```

Comment:
It's weird to define a function inside a static inline function. Why not defining it outside the function?

If you don't want to expose _PyObject_IsImmortal() in the public C API, a better option would be to convert Py_SET_REFCNT() static inline function to a regular function. So you can do whatever you want in the implementation. Calling Py_SET_REFCNT() is uncommon and so doesn't need to be optimized, no?

Hey, it's good to see that my backward incompatible C API change on Py_REFCNT() being useful :-) In Python 3.9, it was allowed to write ``Py_REFCNT(obj) = refcnt;``. I fixed many C extensions to use Py_REFCNT(): see bpo-39573 if you are curious.

Reply:
If making `_Py_SET_REFCNT()` a regular function isn't a problem then I'll definitely take the cleaner approach.